sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosBatchRequestOptions.java

```diff
@@ -196,6 +199,26 @@ Map<String, String> getHeaders() {
         return this.customOptions;
     }
 
+    /**
+     * Sets the custom ids.
+     *
+     * @param keywordIdentifiers the custom ids.
+     * @return the current request options.
+     */
+    public CosmosBatchRequestOptions setKeywordIdentifiers(Set<String> keywordIdentifiers) {
+        this.keywordIdentifiers = keywordIdentifiers;
+        return this;
+    }
+
+    /**
+     * Gets the custom ids.
+     *
+     * @return the custom ids.
+     */
+    public Set<String> getKeywordIdentifiers() {
+        return keywordIdentifiers;
```

Comment:
Ahh - we have getters in public RequestOptions as well - so, I guess wrapping the set in Collections.unmodifiableSet(keywordIdentifiers) needs to happen in public request options.

Reply:
Changed all the public setters to assign a Collections.unmodifiableSet(0